nixos/tests/cagebreak.nix

```diff
@@ -17,56 +17,27 @@ in
   let
     alice = config.users.users.alice;
   in {
+    # Automatically login on tty1 as a normal user:
     imports = [ ./common/user-account.nix ];
+    services.getty.autologinUser = "alice";
+    programs.bash.loginShellInit = ''
+      if [ "$(tty)" = "/dev/tty1" ]; then
+        set -e
 
-    environment.systemPackages = [ pkgs.cagebreak pkgs.wallutils ];
-    services.xserver = {
-      enable = true;
-      displayManager.autoLogin = {
-        enable = true;
-        user = alice.name;
-      };
-    };
-    services.xserver.windowManager.session = lib.singleton {
-      manage = "desktop";
-      name = "cagebreak";
-      start = ''
-        export XDG_RUNTIME_DIR="/run/user/${toString alice.uid}"
-        ${pkgs.cagebreak}/bin/cagebreak &
-        waitPID=$!
-      '';
-    };
+        mkdir -p ~/.config/cagebreak
+        cp -f ${cagebreakConfigfile} ~/.config/cagebreak/config
 
-    systemd.services.setupCagebreakConfig = {
-      wantedBy = [ "multi-user.target" ];
-      before = [ "multi-user.target" ];
-      environment = {
-        HOME = alice.home;
-      };
-      unitConfig = {
-        type = "oneshot";
-        RemainAfterExit = true;
-        user = alice.name;
-      };
-      script = ''
-        cd $HOME
-        CONFFILE=$HOME/.config/cagebreak/config
-        mkdir -p $(dirname $CONFFILE)
-        cp ${cagebreakConfigfile} $CONFFILE
-      '';
-    };
+        cagebreak
+      fi
+    '';
 
-    # Copied from cage:
-    # this needs a fairly recent kernel, otherwise:
-    #   [backend/drm/util.c:215] Unable to add DRM framebuffer: No such file or directory
-    #   [backend/drm/legacy.c:15] Virtual-1: Failed to set CRTC: No such file or directory
-    #   [backend/drm/util.c:215] Unable to add DRM framebuffer: No such file or directory
-    #   [backend/drm/legacy.c:15] Virtual-1: Failed to set CRTC: No such file or directory
-    #   [backend/drm/drm.c:618] Failed to initialize renderer on connector 'Virtual-1': initial page-flip failed
-    #   [backend/drm/drm.c:701] Failed to initialize renderer for plane
-    boot.kernelPackages = pkgs.linuxPackages_latest;
+    hardware.opengl.enable = true;
+    programs.xwayland.enable = true;
+    environment.systemPackages = [ pkgs.cagebreak pkgs.wallutils ];
```

Comment:
I wonder whether we could change wallutils for the wayland protocol info cli tool of emersion (I believe)

Reply:
Yes, that should work. We're already using it for the Sway module but tbh I'd prefer to keep that part out of this PR (due to the re-testing overhead, etc.).